spec/lib/krikri/harvester_spec.rb

```diff
@@ -1,19 +1,61 @@
 require 'spec_helper'
 
 describe Krikri::Harvester do
-  it 'is a havester' do
-    expect(subject).to be_a Krikri::Harvester
-  end
+
+  # Our subject is an instance of a dummy class that mixes in
+  # Krikri::Harvester.
+  let(:klass) { Class.new }
+  subject { klass.include(Krikri::Harvester).new }
 
   context 'with record_ids implemented' do
     before do
-      described_class.send(:define_method, :record_ids) do
-        [1, 2, 3, 4, 5]
-      end
+      expect(subject).to receive(:record_ids).and_return([1, 2, 3, 4, 5])
```

Comment:
I'm not super invested, either.  Your argument that this context should only contain specs that call `record_ids` sounds good. 

The alternative perspective is that this `expect` over specifies implementation details.  Specifically: `allow` means that when `#record_ids` is implemented `#count` should work; `expect` means that `#count` should work *and* call `#record_ids`.

Either works for now, but I suspect `allow` will prove more durable.